src/components/[guild]/RolePlatforms/components/EditRewardAvailabilityModal/EditRewardAvailabilityModal.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  Stack,
+  Text,
+  Tooltip,
+} from "@chakra-ui/react"
+import Button from "components/common/Button"
+import CheckboxColorCard from "components/common/CheckboxColorCard"
+import { Modal } from "components/common/Modal"
+import { Clock, Hash } from "phosphor-react"
+import rewards from "platforms/rewards"
+import { FormProvider, useController, useForm } from "react-hook-form"
+import { PlatformName } from "types"
+import StartEndTimeForm, {
+  AUTO_TIMEFRAME_PLATFORMS,
+  datetimeLocalToIsoString,
+  getShortDate,
+} from "./components/StartEndTimeForm"
+
+export type RolePlatformAvailabilityForm = {
+  capacity?: number
+  startTime?: string
+  endTime?: string
+}
+
+type Props = {
+  isOpen: boolean
+  onClose: () => void
+  platformType: PlatformName
+  claimedCount?: number
+  defaultValues?: RolePlatformAvailabilityForm
+  isLoading?: boolean
+  onDone: (data: RolePlatformAvailabilityForm) => void
+}
+
+const AUTO_SUPPLY_PLATFORMS: Partial<Record<PlatformName, string>> = {
+  UNIQUE_TEXT: "secrets",
+  POAP: "minting links",
+}
+
+const EditRewardAvailabilityModal = ({
```

Comment:
Actually just renamed this file & extracted the startTime/endTime input from it.